pinot-spi/src/main/java/org/apache/pinot/spi/env/VersionedPropertyReader.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.env;
+
+import com.google.common.base.Preconditions;
+import java.io.Reader;
+import org.apache.commons.configuration2.PropertiesConfiguration.PropertiesReader;
+
+
+/**
+ * VersionedPropertyReader extends the PropertiesReader
+ * <p>
+ * Purpose: loads the segment metadata faster
+ *  - by skipping the unescaping of key and
+ *  - parsing the line by splitting based on first occurrence of separator
+ */
+class VersionedPropertyReader extends PropertiesReader {
+
+  public VersionedPropertyReader(Reader reader) {
+    super(reader);
+  }
+
+  @Override
+  protected void parseProperty(final String line) {
+    // skip the regex based parsing of the line content and splitting the content based on first occurrence of separator
+    // getPropertySeparator(), in general returns the PropertiesConfiguration `DEFAULT_SEPARATOR` value i.e. ' = '.
+    String separator = getPropertySeparator();
+    Preconditions.checkArgument(CommonsConfigurationUtils.VERSIONED_CONFIG_SEPARATOR.equals(separator),
+        String.format("versioned property configuration separator should be equal to '%s'",
+            CommonsConfigurationUtils.VERSIONED_CONFIG_SEPARATOR));
+
+    String[] keyValue = line.split(getPropertySeparator());
+    Preconditions.checkArgument(keyValue.length == 2, "property content split should result in key and value");
```

Comment:
This check might fail if the value happen to have the separator

The `line.split()` does regex matching internally, so better use `StringUtils.split()`, and `StringUtils.split()` can take a `max` param to return just two parts to handle the issue said above.

And since you have get `separator` above, you can save the calls of getPropertySeparator() on L48 and L52.